### PR TITLE
Boilerplate update

### PR DESCRIPTION
**Pull request by queses**
Hi! I have made so improvements on your boilerplate; I have added new description, 'build' script, aliases etc. Also i have worked on styling utils and on application structure. I think that this improvements will be useful for Inferno newcomers, allowing them to start development with minimal config efforts. So I will be grateful if you will accept this push request.
Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,28 +29,51 @@
...
     "history": "^4.6.1",
-    "inferno": "^3",
-    "inferno-component": "^3",
-    "inferno-create-element": "^3",
-    "inferno-router": "^3"
+    "inferno": "^3.5.2",
+    "inferno-compat": "^3.5.2",
+    "inferno-component": "^3.5.2",
+    "inferno-create-class": "^3.5.2",
+    "inferno-create-element": "^3.5.2",
```
Comment (Havunen) [inline package.json]: I would not encourage usage of create-element, create-class if you have JSX plugin

**Comment by Havunen** [comment]
Added img is not relevant to inferno

File: .babelrc [hunk 2]
```diff
@@ -1,10 +1,20 @@
...
+      }
+    }],
+    ["es2015", {
+      "loose": true, "modules": false
+    }],
+    "stage-0"
+  ],
+  "plugins": [
+    "syntax-decorators",
+    "transform-decorators-legacy",
```
Comment (Havunen) [inline .babelrc]: using legacy decarators should be avoided?

File: .babelrc [hunk 3]
```diff
@@ -1,10 +1,20 @@
...
-            "presets": [
-                ["es2015", {"loose": true, "modules": false}], "stage-0"
-            ]
-        }
-    },
-    "plugins": ["babel-plugin-syntax-jsx", ["babel-plugin-inferno", {"imports": true}]]
+  "presets": [
+    ["env", {
+      "targets": {
+        "browsers": ["last 2 versions", "safari >= 7" ]
```
Comment (Havunen) [inline .babelrc]: for browsers Inferno supports IE11, safari9, (Safari 7 has issues with event currentTarget)?

File: package.json [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
   "name": "inferno-boilerplate",
-  "version": "0.0.1",
+  "version": "1.0.0",
   "description": "A basic boilerplate to get you started with Inferno + Inferno Router + JSX",
   "main": "src/index.js",
   "scripts": {
-    "start": "webpack-dev-server --quiet --config webpack.conf.js --host 0.0.0.0",
+    "start": "webpack-dev-server --host 0.0.0.0",
+    "dev": "npm run start",
+    "build": "rm -rf ./public && cross-env NODE_ENV=\"production\" webpack",
```
Comment (Havunen) [inline package.json]: webpack -p  for production mode?

File: package.json [hunk 5]
```diff
@@ -24,28 +29,51 @@
   },
   "homepage": "https://github.com/trueadm/inferno-boilerplate#readme",
   "devDependencies": {
+    "autoprefixer": "^7.1.1",
     "babel-core": "^6.24.1",
     "babel-eslint": "^7.2.1",
-    "babel-loader": "^6.4.1",
+    "babel-loader": "^7.0.0",
```
Comment (Havunen) [inline package.json]: babel-plugin-inferno is not tested with babel 7, there was discussion about this in Slack, can you fix it?

File: postcss.config.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+module.exports = (ctx) => ({
+  map: ctx.env === 'development' ? ctx.map : false,
+  plugins: {
+    "autoprefixer": {
+      "browsers": [
+        "last 3 version",
+        "ie >= 10"
```
Comment (Havunen) [inline postcss.config.js]: IE 11

Comment (Havunen) [reply postcss.config.js]: IE10 dead since 2016 - https://www.microsoft.com/en-us/windowsforbusiness/end-of-ie-support

File: public/index.html [hunk 7]
```diff
@@ -1,9 +1,12 @@
...
-</body>
+  <head>
+    <meta charset="utf-8">
+    <link rel="shortcut icon" href="favicon.ico" type="image/x-icon">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <title>Inferno Boilerplate</title>
+  </head>
+  <body>
+    <div id="app"></div>
+  <script type="text/javascript" src="/manifest-1fd08b98b835c985c26b.js"></script><script type="text/javascript" src="/vendor-1fd08b98b835c985c26b.js"></script><script type="text/javascript" src="/app-1fd08b98b835c985c26b.js"></script></body>
```
Comment (Havunen) [inline public/index.html]: what are these hashtags in filenames?

File: src/app.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+ * Main app container 
+ */
+import Component from 'inferno-component'
+import jss from 'jss'
+import preset from 'jss-preset-default'
+
+// Setup JSS to support nesting, vendor-prefixing etc (need to be done just one time):
+jss.setup(preset())
+
+export default class App extends Component {
```
Comment (Havunen) [inline src/app.js]: this can be pure function? Don't extend Component

File: src/views/contributors.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
+import { Link } from 'inferno-router'
+import Component from 'inferno-component'
+
+import Contributor from '~components/contributor.js'
+
+const CONTRIBUTORS = [
+  {
+    name: 'KFlash',
```
Comment (Havunen) [inline src/views/contributors.js]: KFlash is not real contributor.

File: src/components/contributor.js [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import Component from 'inferno-component'
+// import { createStyleSheet } from 'jss'
```
Comment (Havunen) [inline src/components/contributor.js]: commented out code

**Comment by queses** [comment]
Are you talking about `~assets/mointains.jpg`?

**Comment by Havunen** [comment]
Yep

**Comment by Havunen** [comment]
Sorry for late review, I forgot to push submit button...

**Comment by queses** [comment]
Okay, thanks. Think I fixed this issues in latest commit.

**Comment by Havunen** [comment]
Can you fix the merge conflicts please

**Pull request closed without merging**